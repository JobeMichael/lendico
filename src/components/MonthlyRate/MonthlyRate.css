.monthly-rate-value,
.monthly-rate-label {
  font-size: 35px;
  font-weight: normal;
}
.monthly-rate-value {
  font-weight: 700;
}

.error-msg {
  color: red;
}

.error-msg::before {
  display: inline;
  content: "âš  ";
}
