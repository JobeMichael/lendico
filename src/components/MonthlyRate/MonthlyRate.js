import React from "react";
import "./MonthlyRate.css";

const MonthlyRate = ({ monthlyRate, error }) => {
  if (error) {
    return <p className="error-msg">{error}</p>;
  }

  return (
    <>
      {monthlyRate ? (
        <>
          <span className="monthly-rate-label">Monthly Installment :</span>
          <span className="monthly-rate-value">{`${monthlyRate} EUR`}</span>
        </>
      ) : (
        <p data-testid="infoText">
          Enter Amount and duration to calculate the installment
        </p>
      )}
    </>
  );
};

export default MonthlyRate;
