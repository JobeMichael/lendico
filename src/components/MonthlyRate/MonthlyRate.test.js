import { render } from "@testing-library/react";
import React from "react";
import MonthlyRate from "./MonthlyRate";

describe("rendering", () => {
  test("renders MonthlyRate correctly", () => {
    const { asFragment } = render(<MonthlyRate monthlyRate="123" />);

    expect(asFragment()).toMatchSnapshot();
  });

  test("renders information message if no value", () => {
    const { getByTestId } = render(<MonthlyRate />);
    const infoText = getByTestId("infoText");

    expect(infoText.innerHTML).toBe(
      "Enter Amount and duration to calculate the installment"
    );
  });

  test("renders error message correctly", () => {
    const { asFragment } = render(<MonthlyRate error="Error" />);

    expect(asFragment()).toMatchSnapshot();
  });
});
