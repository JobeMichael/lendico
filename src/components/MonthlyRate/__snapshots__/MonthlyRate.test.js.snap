// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders MonthlyRate correctly 1`] = `
<DocumentFragment>
  <span
    class="monthly-rate-label"
  >
    Monthly Installment :
  </span>
  <span
    class="monthly-rate-value"
  >
    123 EUR
  </span>
</DocumentFragment>
`;

exports[`rendering renders error message correctly 1`] = `
<DocumentFragment>
  <p
    class="error-msg"
  >
    Error
  </p>
</DocumentFragment>
`;
